modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsConfigurationManager.java

```diff
@@ -334,6 +339,12 @@ public void stop() {
 
     /** */
     private void scanAndCheckLocalStatistics(AffinityTopologyVersion topVer) {
+        if (!active) {
+            if (log.isDebugEnabled())
+                log.debug("Ignore statistics collection due to inactive cluster state.");
+
+            return;
+        }
```

Comment:
Why is async task below not synchronized with deActivation?

Reply:
Move check into async task.